[ADD] Manufacturing: Dropship subcontracting lead times (#9297)
odoo/documentation: content/applications/inventory_and_mrp/manufacturing/subcontracting/dropship_subcontracting_lead_times.rst

.. |BoM| replace:: :abbr:`BoM (Bill of Materials)`

In Odoo, lead times are used to predict how long it takes to complete a certain action. For example,
a *delivery lead time* can be set for a purchased product, which specifies the number of days it
usually takes for the product's vendor to delivery the product to the purchasing company.

For subcontracted products specifically, delivery lead times can be configured to take into account
the amount of time required for the subcontractor to manufacture a product. Doing so allows the
contracting company to better predict the delivery dates of subcontracted products.

Certain subcontracted products require the contracting company to supply the subcontractor with
manufacturing components. In this case, a *manufacturing lead time* can be used in addition to the
delivery lead time, to generate the date on which the subcontractor must receive the required
components, in order to manufacture the product and deliver it on time.

In cases where components are dropshipped to the subcontractor, an additional *delivery lead time*
can be set for each component, to generate the date that the component must be purchased, so that it

> can be dropshipped to the subcontractor by the date on which manufacturing must begin.

Reviewer: A lot of this makes sense to me, and the explanation is well-written! I am however, still a little lost about the distinction between the product delivery lead time and the component delivery lead time.

Checking my understanding here:
- The product delivery lead time is the duration between confirmation of the sales order of the product to delivery to the customer, right? 
- Component lead time is the duration between placing the PO and its arrival at the subcontractor location, for manufacturing?

It might be helpful for readers to understand the various lead time definitions and how it impacts the subcontracting workflow to have textbook-like definitions of the lead times in one place, and how the lead times are used to schedule various supply chain operations so customers get them in time. This is just a suggestion, coming from someone who's a little confused :sweat_smile:

To set a delivery lead time for a product's subcontractor, navigate to :guilabel:`Inventory app -->
Products --> Products`, and select a subcontracted product.

Then, select the :guilabel:`Purchase` tab on the product's page. If the subcontractor has not
already been added as a vendor, do so now by clicking :guilabel:`Add a line`, and selecting the
subcontractor in the :guilabel:`Vendor` column.

Once the subcontractor has been added, enter the number of days it takes them to manufacture and
deliver the product, in the :guilabel:`Delivery Lead Time` column.

.. image:: dropship_subcontracting_lead_times/delivery-lead-time.png
   :align: center
   :alt: The Delivery Lead Time field for a subcontractor, on the Purchase tab of a product page.

Product manufacturing lead time
-------------------------------

Next, navigate to the product's |BoM| by clicking the :guilabel:`Bill of Materials` smart button at
the top of the product's page. Then, select a BoM from the list.

On the |BoM|, select the :guilabel:`Miscellaneous` tab. In the :guilabel:`Manuf. Lead Time` field,
enter the same number of days that was entered in the :guilabel:`Delivery Lead Time` field of the

> |BoM|'s product.

Reviewer: Just checking: the manufacturing lead time is the same as the product delivery lead time?

I'm slightly confused because in the Mike's Bikes example below, the delivery lead time is 5 days, and the manufacturing lead time is 3 days

.. image:: dropship_subcontracting_lead_times/manufacturing-lead-time.png
   :align: center
   :alt: The Manuf. Lead Time field on a product's BoM.

While not all of these days will actually be used for manufacturing by the subcontractor, setting
the same number of days in each field tells Odoo that the subcontractor must receive the components
and begin production by the start of the product's delivery lead time. This gives the subcontractor
enough time to both manufacture and deliver the product.

Component delivery lead time
----------------------------

From the product's BoM, navigate to each component by clicking on the component in the

> :guilabel:`Components` tab, and then clicking the :guilabel:`➡️ (Internal link)` button to the right

Reviewer: I think @samueljlieber can give the best advice on this, but I think we're moving away from using emojis. Instead, we're hover over the icon, right clicking, and selecting "Inspect" to find the icon name. This one though, I wasn't able to get working.

![image](https://github.com/odoo/documentation/assets/23611711/21138391-af2c-468d-abe3-9945048ca8b2)

It looks like the icon code is "oi-arrow-right", but I couldn't get it working. This is what I tried:
```suggestion
:guilabel:`Components` tab, and then clicking the :icon:`oi-arrow-right` :guilabel:`(Internal link)` button to the right
```

the :guilabel:`Vendor` column.

Once the vendor has been added, enter the number of days it takes them to dropship the product to
the subcontractor, in the :guilabel:`Delivery Lead Time` column.

Dropship subcontracting workflow
================================

Create a request for quotation (RfQ) for the product by navigating to :menuselection:`Purchase app
--> Orders --> Requests for Quotation`, and clicking :guilabel:`New`.

Specify the subcontractor in :guilabel:`Vendor` field. Then, add the product in the
:guilabel:`Products` tab by clicking :guilabel:`Add a product`, selecting the product in the
:guilabel:`Product` column, and specifying a quantity in the :guilabel:`Quantity` column.

In the :guilabel:`Expected Arrival` field, enter a date that provides enough time for the component
vendor to dropship the components, and the subcontractor to manufacture and deliver the product.

.. important::
   When a product is added to an |RfQ|, the *Expected Arrival* field auto-populates with a date that
   is today's date plus the delivery lead time of the product. However, this does not consider the
   time it takes to dropship the components to the subcontractor.

>    When purchasing a product subcontracted using the Dropship Subcontractor on Order route, it is

Reviewer: ```suggestion
   When purchasing a product subcontracted using the *Dropship Subcontractor on Order* route, it is
```

   necessary to adjust this date to take into account the extra time needed for the components to be
   delivered to the subcontractor.

   Since production won't begin until they receive the components, leaving the date as is results in
   the finished product arriving *after* the date listed on the |RfQ|.

Next, click :guilabel:`Confirm Order` to turn the |RfQ| into a |PO|. Doing so creates a second |RfQ|
to purchase the components from the dropshipper and have them sent to the subcontractor.

Navigate to :menuselection:`Purchase app --> Orders --> Requests for Quotation`, and select the
|RfQ| that lists the dropshipper in the :guilabel:`Vendor` column.

On the |RfQ|, the :guilabel:`Expected Arrival` field lists the date on which the subcontractor must

> receive the components, in order to deliver the finished product by the Expected Arrival date listed

Reviewer: ```suggestion
receive the components, in order to deliver the finished product by the :guilabel:`Expected Arrival` date listed
```

accessed from the :guilabel:`Resupply` smart button that now appears on the subcontractor |PO|.

After the dropshipper has delivered the components to the subcontractor, click the
:guilabel:`Validate` button at the top of the dropship order to confirm that the subcontractor has
received the components.

Once the subcontractor receives the components, they begin manufacturing the component, before
delivering it to the contracting company.

.. example::

   Bike retailer *Mike's Bikes* works with a subcontractor — *Bike Friends* — to produce units of
   their *Bicycle* product.

   Mike's Bikes must purchase the required components from vendor Bike World, and have them
   dropshipped to Bike Friends.

   On average, Bike Friends takes three days to manufacture each bicycle, plus an addition two days
   to deliver it to Mike's Bikes.

   As a result, Mike's Bikes sets a delivery lead time of five days for bicycle manufactured by Bike
   Friends: three days for manufacturing, plus two days for delivery.

   On the Bicycle's BoM, they enter a manufacturing lead time of five days as well, to remind
   themselves of the date that components must be delivered to the subcontractor.

>    On the product pages for each of the bicycle's components, they assign Bike World a delivery lead
>    time of two days. This is the amount of time it takes Bike World to dropship each component.

Reviewer: OH I THINK I get it now? This additional lead time is the time it takes the vendor to ship the component to the subcontractor?

In that case, to clarify, can we say something like:
```suggestion
   On the product pages for each of the bicycle's components, they assign Bike World a delivery lead
   time of two days. This is the amount of time it takes Bike World to dropship each component directly to the subcontractor.
```

The :guilabel:`Order Deadline` field lists the latest date on which the |RfQ| can be confirmed, in
order for the dropshipperto deliver the components to the subcontractor by the :guilabel:`Expected
Arrival` date.

Click :guilabel:`Confirm Order` to turn the |RfQ| into a |PO|, and confirm the purchase of the
components from the dropshipper. Doing so causes a :guilabel:`Dropship` smart button to appear at
the top of the page.

Click the :guilabel:`Dropship` smart button to open the dropship order. This order can also be
accessed from the :guilabel:`Resupply` smart button that now appears on the subcontractor |PO|.

After the dropshipper has delivered the components to the subcontractor, click the
:guilabel:`Validate` button at the top of the dropship order to confirm that the subcontractor has
received the components.

Once the subcontractor receives the components, they begin manufacturing the component, before
delivering it to the contracting company.

.. example::

   Bike retailer *Mike's Bikes* works with a subcontractor — *Bike Friends* — to produce units of
   their *Bicycle* product.

   Mike's Bikes must purchase the required components from vendor Bike World, and have them

>    dropshipped to Bike Friends.

Reviewer 2: ```suggestion
   dropshipped to *Bike Friends*.
```

>    On average, Bike Friends takes three days to manufacture each bicycle, plus an addition two days
>    to deliver it to Mike's Bikes.

Reviewer 2: ```suggestion
   On average, *Bike Friends* takes three days to manufacture each *Bicycle*, plus an addition two days
   to deliver it to *Mike's Bikes*.
```

>    As a result, Mike's Bikes sets a delivery lead time of five days for bicycle manufactured by Bike
>    Friends: three days for manufacturing, plus two days for delivery.

Reviewer 2: ```suggestion
   As a result, *Mike's Bikes* sets a delivery lead time of five days for a *Bicycle* manufactured by *Bike
   Friends*: three days for manufacturing, plus two days for delivery.
```

>    On the Bicycle's BoM, they enter a manufacturing lead time of five days as well, to remind

Reviewer 2: ```suggestion
   On the BoM for the *Bicycle*, they enter a manufacturing lead time of five days, as well, to remind
```

   themselves of the date that components must be delivered to the subcontractor.

>    On the product pages for each of the bicycle's components, they assign Bike World a delivery lead

Reviewer 2: ```suggestion
   On the product pages for each of the components for the *Bicycle*, they assign *Bike World* a delivery lead
```

>    time of two days. This is the amount of time it takes Bike World to dropship each component

Reviewer 2: ```suggestion
   time of two days. This is the amount of time it takes *Bike World* to dropship each component
```

   directly to the subcontractor.

>    On May 10th, Mike's Bikes confirms a |PO| for one bicycle, with an expected delivery date of May

Reviewer 2: ```suggestion
   On May 10th, *Mike's Bikes* confirms a |PO| for one *Bicycle*, with an expected delivery date of May
```

   17th.

>    The |RfQ| for purchasing the components from Bike World and having them dropshipped to Bike
>    Friends has an Expected Arrival date of May 12th, and a Deadline of May 10th. The |RfQ| must be
>    confirmed by the deadline in order for Bike Friends to receive the components by the Expected
>    Arrival date, giving them enough time to deliver the finished bicycle to Mike's Bikes by May
>    17th.

Reviewer 2: ```suggestion
   The |RfQ| for purchasing the components from *Bike World*, and having them dropshipped to *Bike
   Friends*, has an *Expected Arrival* date of May 12th, and a Deadline of May 10th. The |RfQ| must be
   confirmed by the deadline in order for *Bike Friends* to receive the components by the *Expected
   Arrival* date, giving them enough time to deliver the finished *Bicycle* to *Mike's Bikes* by May
   17th.
```

   .. image:: dropship_subcontracting_lead_times/deadline-arrival.png
      :align: center
      :alt: The Order Deadline and Expected Arrival dates on a dropship order.

>    Mike's Bikes confirms the |RfQ| on May 10th, and Bike World delivers the components to Bike
>    Friends on May 12th. Bike Friends manufactures the bicycle, and delivers it to Mike's Bikes on
>    May 17th.

Reviewer 2: ```suggestion
   *Mike's Bikes* confirms the |RfQ| on May 10th, and *Bike World* delivers the components to *Bike
   Friends* on May 12th. *Bike Friends* manufactures the *Bicycle*, and delivers it to *Mike's Bikes* on
   May 17th.
```
